src/engineio/transport_new.rs

```diff
@@ -0,0 +1,52 @@
+use crate::error::Result;
+use adler32::adler32;
+use bytes::{Buf, Bytes};
+use std::io::Read;
+use std::io::Write;
+use std::time::SystemTime;
+
+pub trait Transport {
+    /// Sends a packet to the server. This optionally handles sending of a
+    /// socketio binary attachment via the boolean attribute `is_binary_att`.
+    fn emit(&self, data: Bytes, is_binary_att: bool) -> Result<()>;
+
+    /// Performs the server long polling procedure as long as the client is
+    /// connected. This should run separately at all time to ensure proper
+    /// response handling from the server.
+    fn poll(&self) -> Result<Bytes>;
+
+    /// Returns start of the url. ex. http://localhost:2998/engine.io/?EIO=4&transport=polling
+    /// Must have EIO and transport already set.
+    fn base_url(&self) -> Result<String>;
+
+    /// Used to update the base path, like when adding the sid.
+    fn set_base_url(&self, base_url: String) -> Result<()>;
+
+    /// Full query address
+    fn address(&self) -> Result<String> {
+        let reader = format!("{:#?}", SystemTime::now());
+        let hash = adler32(reader.as_bytes()).unwrap();
+        Ok(self.base_url()? + "&t=" + &hash.to_string())
+    }
```

Comment:
Maybe `get_address` is a better name, what do you think?

Reply:
https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

Reply:
Alright good catch!